lib/veri_traci.ml

```diff
@@ -0,0 +1,89 @@
+open Bap.Std
+module Bap_value = Value
+open Bap_traces.Std
+open Monad.State
+open Bil.Result
+
+type event = Trace.event
+
+let mem_of_arch arch = 
+  let (module T : Target) = target_of_arch arch in
+  T.CPU.mem
+
+(** TODO: what is a behaviour of update_event method?  *)
+class context arch = object(self:'s)
+  inherit Bili.context
+  val mem  : var = mem_of_arch arch
+  val arch : arch = arch
+  method arch = arch
+  method mem = mem
+  method update_event: event -> 's = fun ev -> failwith "unimplemented"
+end
+
+let mem_operands mv = 
+  Size.of_int_opt (Bitvector.bitwidth (Move.data mv)) |> function
+  | None -> None
+  | Some sz ->
+    let addr = Bil.int (Move.cell mv) in
+    let data = Bil.int (Move.data mv) in
+    Some (addr, data, sz)
+
+class ['a] t = object(self)
```

Comment:
Context of execution is a set of things that changes during the excution, e.g., memory, registers, cpu state. Architecture, is definitely not going to change and is not a part of the context. What you're doing is choosing a very long way to pass an arch parameter to the interpreter. It is ok, and kind of expected, for the interpreter to depend on architecture. It is nice, if your context doesn't depend on the architecture. And it in fact doesn't depend.  If there is some architecture specific behavior, then it should be implemented in the context. 
So, please, move arch to interpreter's parameters.